core-bundle/src/Search/Backend/Provider/TableDataContainerProvider.php

```diff
@@ -86,18 +88,23 @@ public function updateIndex(ReindexConfig $config): iterable
 
     public function convertDocumentToHit(Document $document): Hit|null
     {
-        // TODO: service for view and edit URLs
-        $viewUrl = 'https://todo.com?view='.$document->getId();
-        $editUrl = 'https://todo.com?edit='.$document->getId();
-
         $row = $this->loadRow($this->getTableFromDocument($document), (int) $document->getId());
 
         // Entry does not exist anymore -> no hit
         if (false === $row) {
             return null;
         }
 
-        $title = $this->recordLabeler->getLabel(\sprintf('contao.db.%s.id', $this->getTableFromDocument($document)), $row);
+        $table = $this->getTableFromDocument($document);
+        $editUrl = $this->dcaUrlAnalyzer->getEditUrl($table, (int) $document->getId());
+        $viewUrl = $this->dcaUrlAnalyzer->getViewUrl($table, (int) $document->getId());
```

Comment:
/cc @Toflar - This happens before `isHitGranted()` is called and throws the AccessDeniedException

Should this be wrapped with a tryCatch?

Reply:
Creating the Hit should not happen before checking access to the document.

Reply:
try/catch still makes sense though I think? Added in ea79e48d191c7f47a3fe7cd3b3be4a3452f09713

Reply:
Ah, I see. Now I remember. The voting happens on the `Hit`, not on the `Document`. The reason why I designed this like it is now is that there can only be one provider knowing how to convert a search document to a `Hit`. It is the one also providing the documents for indexing. But there might be multiple listeners that want to modify the way a `Hit` is formatted (or hide it) depending on some third party settings.
However, I'm not entirely sure this is correct. We have an `IndexDocumentEvent` and an `EnhanceHitEvent` - so indexing can be modified on the `Document` (which is correct) but reading/querying only on the `Hit`. I don't think it's wrong to have an event to be able to modify the `Hit` but maybe the permissions should also be checked on the `Document` instead of the `Hit` - wdyt?

Reply:
It already kinda shows because the `ProviderDelegatingVoter` works on the document already:

```
foreach ($this->providers as $provider) {
            if ($provider->supportsType($subject->getDocument()->getType())) {
                return true;
            }
        }
```

Maybe I should change that do `isDocumentGranted()` - then the error encountered should not happen anymore. Makes no sense to convert a `Document` to a `Hit` when you don't have access to it.

Reply:
There we go: https://github.com/contao/contao/pull/7866

Reply:
With https://github.com/contao/contao/pull/7866 being merged now I think we could ignore the try-catch because it shouldn't happen but leaving it in doesn't harm either. Marking as resolved.